src/rendergraph/shaders/generated_shaders_gl.cmake

```diff
@@ -0,0 +1,13 @@
+set(
+  generated_shaders_gl
+  pattern.vert.gl
+  rgb.vert.gl
+  rgba.vert.gl
+  texture.vert.gl
+  unicolor.vert.gl
+  pattern.frag.gl
+  rgb.frag.gl
+  rgba.frag.gl
+  texture.frag.gl
+  unicolor.frag.gl
+)
```

Comment:
How would you feel about automating the generation of the `.gl` script using something like this?

```cmake
execute_process(
  COMMAND ./generate_shaders_gl.pl
  RESULT_VARIABLE result_generate_gl
)
if (${result_generate_gl})
  message(FATAL_ERROR "Unable to generate the GL files: ${result_generate_gl}")
endif()
```

Alternatively, what do you think of auto generating this in the `pre-commit` hooks? (Note: using Python for the script might be a requirement, but this needs to be checked)

```yaml
      - id: generate-rendergraph-gl
        name: generate-rendergraph-gl
        description: "Generate GL files for rendergraph saher"
        entry: ./src/rendergraph/shaders/generate_shaders_gl.pl
        require_serial: true
        language: perl
        files: ^src/rendergraph/shaders/pattern.frag/.*\.(frag|vert)$
```

Reply:
I don't know if it is worth the effort. I can simply commit the generated shaders. If the need arises to add more shaders, we can improve on this. Also note that this is only a temporary solution: With Qt >=6.6 we can use the qsb bundles from C++ directly.

Reply:
As we've just updated to Ubuntu 24.04 (Qt 6.4), it may be 2 years+ before the minimum Qt version is 6.6+. I don't mind updating this in a follow up PR tho.

Reply:
Thanks, yes, doing this in a follow up PR would be best.

Reply:
pre-commit is the better option, as it will only run on changes, not every time you build. Also, I'm not sure that python is already a build-time dependency.

I included the pre-commit hook in https://github.com/m0dB/mixxx/pull/18